tt_metal/impl/device/device.cpp

```diff
@@ -2983,7 +2983,14 @@ bool Device::initialize(const uint8_t num_hw_cqs, size_t l1_small_size, size_t t
     this->using_fast_dispatch = false;
     this->num_hw_cqs_ = num_hw_cqs;
     constexpr uint32_t harvesting_map_bits = 12;
-    this->build_key_ = ((uint32_t)this->num_hw_cqs_ << harvesting_map_bits) | tt::Cluster::instance().get_harvesting_mask(this->id());
+    this->build_key_ = ((uint32_t)this->num_hw_cqs_ << harvesting_map_bits);
+    if (not hal.is_coordinate_virtualization_enabled()) {
+        // Coordinate virtualization is not enabled. For a single program, its associated binaries will vary across devices with different cores harvested.
+        this->build_key_ = (this->build_key_) | tt::Cluster::instance().get_harvesting_mask(this->id());
+    } else {
+
+        this->build_key_ = (this->build_key_) | (std::bitset<12>(tt::Cluster::instance().get_harvesting_mask(this->id())).count());
```

Comment:
I would include your comment in your PR description:

> Modify the build key to only account for the number of rows harvested, and not the exact harvesting mask for architectures with Coordinate Virtualization enabled

Reply:
Added a description